### ISSUE TITLE
Hide comments/projects/studios by specific users

### ISSUE THREAD

**Issue by DogCatPuppyLover** [ScratchAddons/ScratchAddons#3810]
### Is your request related to a problem? Please describe.

If someone spams or harasses another user, the most the harassed user can often do is report them. This is often not helpful, however, as unless multiple users report the comment, they have to wait until a moderator reviews the report. This can make Scratch unpleasant for some users, especially those who get harassed often. The Scratch Team has rejected a feature to block users on the basis that this would prevent people from reporting inappropriate content, but I believe this would be a useful feature for Scratch Addons.

### Describe the solution you'd like

I would like an addon that hides comments, projects, and studios from a list of users specified by the SA user. Actions by that user will not show up in the user's messages, and projects and studios by a blocked user will show a page with a button to unblock the user.

There could also be an option to automatically delete all new comments by blocked users on the SA user's profile or projects. This would be more work to create, but could work by checking for new comments every few minutes and deleting ones by blocked users.

### Describe alternatives you've considered

If someone is harrassing you on your profile or projects, you can delete their comments or toggle commenting. However, the former requires you to view all the messages from the user, and the latter restricts commenting for other users as well.

**Comment by apple502j** [comment]
TL;DR: while I understand its use case, the implementation you mentioned could cause problems that we don't really want to have.

Hi, @DogCatPuppyLover - thanks for a new suggestion! And sadly yeah, this is a valid suggestion. I understand that the moderation team in Scratch is very underpowered for the current spike in users, and I am sure Scratch is not 100% safe place from transphobia, etc given how the world (and the rest of the internet) is right now. Of course we do not moderate the website so there may be some complicated stuff, but hey, Scratch Addons is for that exact reason!

However, adding a feature to Scratch Addons is not that easy. We first have to check if it's technically even possible/plausible. For comments and projects it could be, since in most cases those are displayed along with its author. This may cause a performance drop on pages like the front page and profiles, but that should be fine.

Studios aren't that easy; a studio host information is unavailable unless we make an API call to that page. That'd be too spammy and potentially disruptive to the server.

The thing I'm most concerned about is the automated comment deletion - since there are always chances the comment is talking about a legitimate thing, like an apology. Deleting such comments could be seen as a rude action. Also, there should be a way to report them (not just hide/delete) in case the Scratch Team actually sees them - but automating that is too powerful.

Do you have any ideas or suggestions on this?

\- apple502j (nb, aroace; they/them)

(for other contributors: this is a great opportunity to retract and remove your homophobic/transphobic forum posts and ocular reactions. we don't want that.)

**Comment by WorldLanguages** [comment]
Automatically deleting comments: no
A feature that would probably be perfectly fine would be to blur the contents of comments by users you've blocked, until you click on the comment. But I'm not sure how useful that would be.

**Comment by DogCatPuppyLover** [comment]
It would probably work fine to only hide comments and notifications ([user] loved your project, etc.) by blocked users, as most harassment does happen through comments. Projects and userpages could also be hidden, but those are less important. I wasn't talking specifically about transphobia, but that is an issue and I have gotten transphobic messages.

I agree about automated comment deleting or reporting; that could also get users banned by the ST for using a "bot." I think that an option to blur comments or completely hide them would be useful, as users could easily un-hide blurred comments when they're ready and report them. It might not be very useful to just blur comments, however, as it would probably be hard to resist the temptation to click on them.

**Comment by Samq64** [comment]
![Screenshot 2021-11-15 at 09-47-10 Scratch - Imagine, Program, Share](https://user-images.githubusercontent.com/81489795/141801954-7347ea01-f493-403e-8e8f-27ee48fa13a8.png)

**Comment by mxmou** [comment]
> (for other contributors: this is a great opportunity to retract and remove your homophobic/transphobic forum posts and ocular reactions. we don't want that.)

Asking this again: who are the "we" who don't want it?

**Comment by DNin01** [comment]
From a Chrome Web Store review (with 42 upvotes):
> Can you please add a option where you can block people because there is someone being rude to me and I don't want to see them again.

**Comment by WorldLanguages** [comment]
Yes, this is frequently requested.

But I don't know what to think, if this was a good idea it wouldn't be a rejected suggestion by the Scratch Team
